go/apis/kube-mail/v1alpha1/emailpolicy_types.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type EmailPolicySMTPSink struct {
+	Server      v1.ObjectReference `json:"server"`
+	Credentials v1.ObjectReference `json:"credentials"`
+}
+
+type EmailPolicyRateLimiting struct {
+	Maximum int    `json:"maximum"`
+	Period  string `json:"period"`
+}
+
+type EmailPolicySink struct {
+	SMTP EmailPolicySMTPSink `json:"smtp"`
+}
+
+type EmailPolicySpec struct {
+	Default      bool                    `json:"default"`
+	PodSelector  metav1.LabelSelector    `json:"podSelector"`
+	RateLimiting EmailPolicyRateLimiting `json:"ratelimiting"`
+	Sink         EmailPolicySink         `json:"sink"`
+}
```

Comment:
Having the types defined in both Go and TS is just an accident waiting to happen. Is there any conceivable way to have _one_ authorative definition that all other representations are automatically generated from?

Reply:
I can now generate the TypeScript types and CRDs from the go definition. I guess this is the best we can do.
It also has the benefit of using `controller-gen` to create our CRDs, like all other go based operators do.